### ISSUE TITLE
Feature Request: Pallete Animation for Colour Cycling

### ISSUE THREAD

**Issue by TheConfuZzledDude** [aseprite/aseprite#1067]
Way back in the day when space was limited, a cool trick to have really smooth animation of environments while maintaining disk space was to instead animate the palette, cycling the colours to create smooth transitions and animations without wasting space. It was used for everything from flowing water, to changing lighting, and even rain and snow.

[An example](http://www.effectgames.com/demos/canvascycle/?sound=0)

Basically this would mean attaching the current palette (only with indexed images I would suppose) to the frame of the animation, rather than it being global. This may also work well with the palette gradient feature, but instead changing the gradient across frames.

**Comment by dacap** [comment]
Hi @TheConfuZzledDude, it's strange but I thought that this issue was duplicated. At least in my head I've this already planned (internally a sprite can have multiple color palettes, but there is no User Interface to do that).

**Comment by davidpgil** [comment]
I’d love to see this too. Been looking for a software to replace Pro Motion that is free and open source thst can do this.

**Comment by dacap** [comment]
There are some work-in-progress about this one, internally it's supported, but the most difficult part is the UI (which needs some redesign of the Timeline to show the palette changes like a new layers)

**Comment by robin-raymond** [comment]
Any updates on if/when this feature might happen?

**Comment by XCompWiz** [comment]
There's very few references of this in the world now that I can find, but I wanted to mention two.
http://www.effectgames.com/demos/canvascycle/ has nice examples and lets you see the palette animation in progress.
Mark Ferrari, one of the pioneers of the technique, talks about how he's done it in various tools: https://www.youtube.com/watch?v=aMcJ1Jvtef0

**Comment by dacap** [comment]
Related to:
https://github.com/aseprite/aseprite/issues/1731#issuecomment-546699948
https://community.aseprite.org/t/color-cycling-effect/4137/3?u=dacap

**Comment by iangilman** [comment]
FYI, the community of artists who are fans of the color cycling artwork in my app (https://pixfabrik.com/livingworlds/) are excited for the possibility of this feature! Once it's in, I'll be sure to spread the word 😊

**Comment by Axis4s** [comment]
Big 2025, any news on this feature?

**Comment by dacap** [comment]
Bug reported in:
* https://www.reddit.com/r/aseprite/comments/1rloctj/comment/o8tosn3/
* internal: https://igarastudio.zendesk.com/agent/tickets/9007
due to internals used in Aseprite: loading a sequence of indexed color .png files will create palette changes in the timeline, but then handling/adding/removing frames will break the time/frame of those palettes.

We need the UI for this ASAP.
